rad/prelude/auth.rad

```diff
@@ -0,0 +1,85 @@
+{:module 'prelude/auth
+ :doc "Functions for easily handling authorization on chains."
+ :exports
+ '[make-simple-auth]}
+
+(import prelude/validation :as 'validation)
+(import prelude/set :as 'set)
+(import prelude/dict '[dict-from-seq keys] :unqualified)
+(import prelude/patterns :unqualified)
+(import prelude/ref '[modify-ref] :unqualified)
+
+(def make-simple-auth
+  "Creates an authorization scheme with any number of groups of users.
+
+  Example:
+
+    (def auth (make-simple-auth [:admin])) ;; creates auth with initial 'admin group'
+    (auth [:add-to :admin \"alice\"]) ;; adds alice to admin
+    (auth [:is-in :admin \"alice\"]) :: #t
+    (auth [:create-group :regular]) ;; creates 'regular' group
+
+  A common pattern is to set up a machine, and only then set the owner of the
+  machine. Since setting the owner is something only owners can do, we provide
+  a way for *initializing* the authentication mechanism. Prior to
+  initialization, `(auth [:allowed <group> <person>])` always succeeds.
+  After `(auth [:initialize])` is called, it returns whatever
+  `(auth [:is-in <group> <person>])` returns.
+"
+  (fn [set-names]
+    (def groups (ref (dict-from-seq (map (fn [k] [k (ref set/empty)]) set-names))))
```

Comment:
Why do we need refs at the keys?

Reply:
Don't need it, but feels slightly easier to modify.